caas/kubernetes/provider/provider.go

```diff
@@ -50,7 +51,7 @@ func (kubernetesEnvironProvider) Open(args environs.OpenParams) (caas.Broker, er
 	if err := validateCloudSpec(args.Cloud); err != nil {
 		return nil, errors.Annotate(err, "validating cloud spec")
 	}
-	broker, err := NewK8sBroker(args.Cloud, args.Config, newK8sClient)
+	broker, err := NewK8sBroker(args.Cloud, args.Config, newK8sClient, nil, jujuclock.WallClock)
```

Comment:
I guess we could pass in newKubernetesWatcher here instead of nil, and remove the nil check in NewK8sBroker. I think that would be cleaner?

Reply:
agreed. changed.